snuba/datasets/errors_replacer.py

```diff
@@ -565,6 +614,22 @@ def pre_replacement(self, replacement: Replacement, matching_records: int) -> bo
 
         return False
 
+    def post_replacement(self, replacement: Replacement, matching_records: int) -> None:
+        """
+        Write the offset just processed to Redis if execution took longer than the threshold.
+        """
+        if (
+            time.time() - self.__start_time
+        ) < settings.REPLACER_PROCESSING_TIMEOUT_THRESHOLD:
+            return
```

Comment:
This time is in seconds, not milliseconds, but you set your threshold in milliseconds.